src/android/aidl/birthday_service/src/lib.rs

```diff
@@ -26,4 +30,64 @@ impl IBirthdayService for BirthdayService {
     fn wishHappyBirthday(&self, name: &str, years: i32) -> binder::Result<String> {
         Ok(format!("Happy Birthday {name}, congratulations with the {years} years!"))
     }
+    // ANCHOR_END: IBirthdayService
+
+    fn wishWithInfo(&self, info: &BirthdayInfo) -> binder::Result<String> {
+        Ok(format!(
+            "Happy Birthday {}, congratulations with the {} years!",
+            info.name, info.years,
+        ))
+    }
+
+    fn wishWithProvider(
+        &self,
+        provider: &Strong<dyn IBirthdayInfoProvider>,
+    ) -> binder::Result<String> {
+        Ok(format!(
+            "Happy Birthday {}, congratulations with the {} years!",
+            provider.name()?,
+            provider.years()?,
+        ))
+    }
+
+    fn wishWithErasedProvider(
+        &self,
+        provider: &SpIBinder,
+    ) -> binder::Result<String> {
+        use binder::binder_impl::Proxy;
+
+        // Convert the `IBinder` to a concrete interface.
+        let provider = BpBirthdayInfoProvider::from_binder(provider.clone())?;
```

Comment:
I'm not sure i understand your suggestion. `IBirthdayInfoProvider` is a trait, so there's no way to impl `TryFrom` for it. I tried doing `BpBirthdayInfoProvider::try_from` in case that's what you meant, but the compiler says there's no `TryFrom` impl for that type.

Reply:
I saw `FromIBinder::try_from(self)` used at https://android.googlesource.com/platform//frameworks/native/+/refs/heads/main/libs/binder/rust/src/proxy.rs#108, but maybe that should have been `I::try_from`.

Alternatively, you can use the `into_interface` method defined there. See https://android.googlesource.com/platform/system/tools/aidl/+/refs/heads/main/tests/rust/test_client.rs#1054 for an example.

Reply:
`into_inteface` worked! So that unnecessary import is gone.